sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -250,10 +252,10 @@ struct _pi_context : _pi_object {
   // Keep the PI devices this PI context was created for.
   std::vector<pi_device> Devices;
 
-  // If devices in the context are sub-devices of the same device, we want to
-  // save a root-device.
-  pi_device RootDevice = nullptr;
-  bool isContextOfSubDevices() { return RootDevice != nullptr; }
+  // If context contains one device or sub-devices of the same device, we want
+  // to save this device.
+  pi_device SingleRootDevice = nullptr;
+  bool hasSingleRootDevice() { return SingleRootDevice != nullptr; }
```

Comment:
NIT: I'd not add this function at all and just checked `SingleRootDevice` directly